Minesweeper.py

```diff
@@ -43,68 +28,86 @@
+        x -= 1
     elif direction == "right":
-        player_x += 1
-
-    # Check if the player has hit a mine
-    if board[player_y][player_x] == -1:
-        game_over = True
-        print("Game over! You hit a mine.")
-    else:
-        moves += 1
-
-    # Check if the player has won
-    if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES:
-        game_over = True
-        print("Congratulations! You found all the mines.")
-
-    # Place a mine
-    if action == "move":
-        # Do nothing
-        pass
-    elif action == "flag":
-        if board[player_y][player_x] == 0:
-            board[player_y][player_x] = "F"
-            flags += 1
-    elif action == "unflag":
-        if board[player_y][player_x] == "F":
-            board[player_y][player_x] = 0
-            flags -= 1
-
-# Place the mines randomly
-mine_count = 0
-while mine_count < NUM_MINES:
-    x = random.randint(0, BOARD_SIZE - 1)
-    y = random.randint(0, BOARD_SIZE - 1)
-    if board[y][x] == 0:
-        board[y][x] = -1
-        mine_count += 1
-
-# Calculate the number of mines around each cell
-for y in range(BOARD_SIZE):
-    for x in range(BOARD_SIZE):
-        if board[y][x] != -1:
-            count = 0
-            for dy in [-1, 0, 1]:
-                for dx in [-1, 0, 1]:
-                    nx = x + dx
-                    ny = y + dy
-                    if 0 <= nx < BOARD_SIZE and 0 <= ny < BOARD_SIZE and board[ny][nx] == -1:
-                        count += 1
-            board[y][x] = count
\ No newline at end of file
+        x += 1
+    return x, y
+
+def place_mines():
+    mine_count = 0
+    while mine_count < NUM_MINES:
+        x = random.randint(0, BOARD_SIZE - 1)
+        y = random.randint(0, BOARD_SIZE - 1)
+        if board[y][x] == 0:
+            board[y][x] = -1
+            mine_count += 1
+
+def count_nearby_mines(x, y):
+    count = 0
+    for dy in [-1, 0, 1]:
+        for dx in [-1, 0, 1]:
+            nx = x + dx
+            ny = y + dy
+            if 0 <= nx < BOARD_SIZE and 0 <= ny < BOARD_SIZE and board[ny][nx] == -1:
+                count += 1
+    return count
+
+def calculate_mine_counts():
+    for y in range(BOARD_SIZE):
+        for x in range(BOARD_SIZE):
+            if board[y][x] != -1:
+                board[y][x] = count_nearby_mines(x, y)
+
+def game_loop():
+    global player_pos, game_over, moves, flags, mines_found
+
+    while not game_over:
+        clear_screen()
+        print_board()
+        print_stats()
+
+        action = input("Enter your action (move, flag, or unflag): ")
+        direction = input("Enter the direction (up, down, left, or right): ")
+
+        new_pos = update_player_pos(direction)
+        if board[new_pos[1]][new_pos[0]] == -1:
+            game_over = True
+            print("Game over! You hit a mine.")
+        else:
+            player_pos = new_pos
+            moves += 1
+
+        if moves == BOARD_SIZE * BOARD_SIZE - NUM_MINES:
+            game_over = True
```

Comment:
Consider handling the case when the board is completely cleared by providing a separate message instead of using the same success message as when all mines are found. This will improve user feedback in the game. ```diff +            if board[y][x] == 0:  # Ensure only moving on cleared cells +                print("You won! All safe spots cleared!") ``` ---